src/operations.jl

```diff
@@ -13,80 +13,105 @@ gamutmax(::Type{T}) where {T<:HSL} = (360,1,1)
+    Cmin = C(gamutmin(C)...)
+    mapc((m, n) -> T((m - n) * rand(r, floattype(T)) + n), Cmax, Cmin)
+end
+function rand(r::AbstractRNG, ::SamplerType{C}) where {T<:Rand01Type, C0<:AbstractGray{T},
+                                                       C<:Union{C0, TransparentGray{C0, T}}}
+    mapc(_ -> rand(r, T), base_colorant_type(C)())
+end
+function rand(r::AbstractRNG, ::SamplerType{C}) where {T<:Rand01Type, C0<:AbstractRGB{T},
+                                                       C<:Union{C0, TransparentRGB{C0, T}}}
+    mapc(_ -> rand(r, T), base_colorant_type(C)())
+end
+function rand(r::AbstractRNG, ::SamplerType{AGray32}) # Gray24 has little benefit of specialization.
+    reinterpret(AGray32, (rand(r, UInt32) & 0xff0000ff) * 0x010101)
 end
+rand(r::AbstractRNG, ::SamplerType{RGB24}) = reinterpret(RGB24, rand(r, UInt32) & 0xffffff)
+rand(r::AbstractRNG, ::SamplerType{ARGB32}) = reinterpret(ARGB32, rand(r, UInt32))
 
-if Base.VERSION < v"1.6.0-DEV.1083"
-    reinterpretc(::Type{T}, A::Array{C}) where {T<:Real,C<:AbstractGray} = reinterpret(T, A)
-    reinterpretc(::Type{T}, A::Array{C}) where {T<:Real,C<:Colorant} = reshape(reinterpret(T, A), (sizeof(C)÷sizeof(T), size(A)...))
-else
-    reinterpretc(::Type{T}, A::Array{C}) where {T<:Real,C<:Colorant} = reinterpret(reshape, T, A)
+function rand(r::AbstractRNG, ::Type{C}, dims::Dims) where {C <: Colorant}
+    CC = isconcretetype(C) ? C : base_colorant_type(C){Float64}
+    rand!(r, Array{CC}(undef, dims), CC)
 end
 
-function _rand(::Type{T}, sz::Dims) where T<:Colorant
-    Mi = eltype(T) <: FixedPoint ? 1.0/typemax(eltype(T)) : 1.0
-    A = Array{T}(undef, sz)
-    Tr = eltype(T) <: FixedPoint ? FixedPointNumbers.rawtype(eltype(T)) : eltype(T)
-    nchannels = sizeof(T)÷sizeof(eltype(T))
-    Au = Random.UnsafeView(convert(Ptr{Tr}, pointer(A)), length(A)*nchannels)
-    rand!(Au)
-
-    Gmax = gamutmax(T)
-    Gmin = gamutmin(T)
-    s = (Gmax .- Gmin) .* Mi
-    if !all(iszero, Gmin) || !all(==(1), s)
-        Ar = reinterpretc(eltype(T), A)
-        if T<:AbstractGray
-            s1, offset1 = s[1], Gmin[1]
-            for I in CartesianIndices(A)
-                Ar[I] = Ar[I] * s1 + offset1
-            end
-        else
-            for I in CartesianIndices(A)
-                for j in eachindex(s)
-                    Ar[j,I] = Ar[j,I] * s[j] + Gmin[j]
-                end
-            end
-        end
+# rand!
+function rand!(r::AbstractRNG, A::Array{C}, ::SamplerType{C}) where {C<:Colorant}
+    rand!(r, A, SamplerType{base_colorant_type(C){Float64}}())
+end
+function rand!(r::AbstractRNG, A::Array{C}, ::SamplerType{C}) where {T, C<:Colorant{T}}
+    A .= rand.((r,), C)
+end
+function _rand01!(r::AbstractRNG, A::Array{C},
+               ::SamplerType{C}) where {T<:Rand01Type, C<:Colorant{T}}
+    N = sizeof(C) ÷ sizeof(T)
+    T0 = T <: FixedPoint ? FixedPointNumbers.rawtype(T) : T
+    At = unsafe_wrap(Array, reinterpret(Ptr{T0}, pointer(A)), (N, size(A)...))
```

Comment:
Are you using `unsafe_wrap` to bypass the performance problem of `ReinterpretArray` prior to Julia 1.6? I'm fine with that, I just remember someone complaining about that solution previously.

Reply:
No, it has nothing to do with `ReinterpretArray` here. (Of course, therein lies the root problem.)
It is an issue with using the private API.
https://github.com/JuliaLang/julia/issues/36718